### PR TITLE
natron: 2.1.9 -> 2.3.15

### PR DESCRIPTION
**Pull request by vojta001**
###### Motivation for this change
I fixed Natron build and tried it with the [Spaceship example](https://sourceforge.net/projects/natron/files/Examples) and it worked. I am however unsure about enabling the shiboken build for Python 2. It works for me, but I don't know whether I am supposed to do it this way.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> I am however unsure about enabling the shiboken build for Python 2. It works for me, but I don't know whether I am supposed to do it this way.

We are not building python 2 by default anyway so if it works it works and if not you would need to supply a patch.

File: pkgs/applications/video/natron/default.nix [hunk 1]
```diff
@@ -88,9 +95,10 @@ stdenv.mkDerivation {
   inherit version;
   name = "natron-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:96]:
```suggestion
  pname = "natron";
```

File: pkgs/applications/video/natron/default.nix [hunk 2]
```diff
@@ -1,14 +1,14 @@
 { lib, stdenv, fetchurl, qt4, pkg-config, boost, expat, cairo, python2Packages,
   cmake, flex, bison, pango, librsvg, librevenge, libxml2, libcdr, libzip,
   poppler, imagemagick, openexr, ffmpeg_3, opencolorio, openimageio,
-  qmake4Hook, libpng, libGL, lndir }:
+  qmake4Hook, libpng, libGL, lndir, libraw, openjpeg_2, libwebp, fetchgit }:
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:4]:
```suggestion
  qmake4Hook, libpng, libGL, lndir, libraw, openjpeg, libwebp, fetchgit }:
```

File: pkgs/applications/video/natron/default.nix (hunk 1 above)
Comment (vojta001) [reply pkgs/applications/video/natron/default.nix]: Shall I also change `name = "openfx-${pluginName}-${version}";` (line 25) to `pname = "openfx-${pluginName}";`?

Comment (SuperSandro2000) [reply pkgs/applications/video/natron/default.nix]: Yes, please if it uses mkDerivation.

File: pkgs/applications/video/natron/default.nix [hunk 3]
```diff
@@ -65,32 +70,35 @@ let
...
       ];
     })
     ({
       pluginName = "misc";
-      sha256 = "02h79jrll0c17azxj16as1mks3lmypm4m3da4mms9sg31l3n82qi";
+      sha256 = "XkdQyWI9ilF6IoP3yuHulNUZRPLX1m4lq/+RbXsrFEQ=";
       buildInputs = [
         libGL
       ];
       preConfigure = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:86]:
```suggestion
      postPatch = ''
```

File: pkgs/applications/video/natron/default.nix [hunk 4]
```diff
@@ -65,32 +70,35 @@ let
...
 in
 stdenv.mkDerivation {
   inherit version;
-  name = "natron-${version}";
+  pname = "natron";
 
-  src = fetchurl {
-    url = "https://github.com/MrKepzie/Natron/releases/download/${version}/Natron-${version}.tar.xz";
-    sha256 = "1wdc0zqriw2jhlrhzs6af3kagrv22cm086ffnbr1x43mgc9hfhjp";
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:98]: Please use fetchFromGitHub.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/natron/default.nix (hunk 4 above)
Comment (vojta001) [reply pkgs/applications/video/natron/default.nix:98]: I can't. I need submodules to be downloaded as well.

Comment (SuperSandro2000) [reply pkgs/applications/video/natron/default.nix:98]: There is an option for that.

**Comment by SuperSandro2000** [comment]
You're commits should follow the contributing guide.

File: pkgs/applications/video/natron/default.nix (hunk 4 above)
Comment (vojta001) [reply pkgs/applications/video/natron/default.nix]: Didn't know it. Fixed

**Comment by vojta001** [comment]
I thought it will be squashed during merging/rebasing

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113314` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>fmbt</li>
<li>natron</li>
<li>qweechat</li>
<li>yubikey-neo-manager</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
fmbt:
<details>


`lgpl21` is a deprecated license, check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/tools/fmbt/default.nix:48:5:

```
   |
48 |     license = licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
natron:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/video/natron/default.nix:135:5:

```
    |
135 |     license = lib.licenses.gpl2;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
qweechat:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
yubikey-neo-manager:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/misc/yubikey-neo-manager/default.nix:14:15:

```
   |
14 |   patches = [ ./fix-pyside-requirement.diff ];
   |               ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
</details>

File: pkgs/development/python-modules/pyside/shiboken.nix [hunk 5]
```diff
@@ -18,7 +18,6 @@ buildPythonPackage rec {
   pname = "pyside-shiboken";
   version = "1.2.4";
   format = "other";
-  disabled = !isPy3k;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyside/shiboken.nix:21]: Please put this into a separate commit. Is this an accident or did you test that it now works for python3?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pyside/shiboken.nix (hunk 5 above)
Comment (vojta001) [reply pkgs/development/python-modules/pyside/shiboken.nix:21]: I don't know whether it works in Python3. I just found out it is required as a dependency of Natron that requires Python2 and it works out of the box.

Extracted into #115081.

**Comment by vojta001** [comment]
I rebased onto 2a1bda47f7a3365daa6c73dd86d1d28c02f526db thanks to #115081 getting merged.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113314` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>natron</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
natron:
<details>

build log of '/nix/store/bh3cydaqcvqgx6lbs2xbxj6hwvgb1f0s-natron-2.3.15' is not available
error: --- Error --- nix
build log of '/nix/store/bh3cydaqcvqgx6lbs2xbxj6hwvgb1f0s-natron-2.3.15' is not available
When evaluating attribute ‘natron’:
warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/video/natron/default.nix:135:5:

```
    |
135 |     license = lib.licenses.gpl2;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>
